File: 해시/지후 [해시]/완주하지_못한_선수.js
function solution(participant, completion) {
    // 시간 복잡도 문제
    // for문으로 변경
    const sortParticipant = participant.sort();
    const sortCompletion = completion.sort();

    for (let i = 0; i <= sortParticipant.length; i++) {
        if (sortParticipant[i] !== sortCompletion[i]) {
            return sortParticipant[i];
        }
    }
}

File: 해시/[지현] 해시/완주하지_못한_선수.py
def solution(participant, completion):
    answer = ''
    participant.sort()
    completion.sort()
    for i in range(len(participant)):
        if i> len(completion)-1:
            answer=participant[i]
        elif participant[i] !=completion[i]:
            answer=participant[i]
            break

    return answer

# 값은 맞지만 효율성 테스트에서 시간초과 남
#def solution(participant, completion):
#     answer = ''
#     participant.sort()
#     completion.sort()
#     for i in participant:
#         if completion==[]:
#             answer=i
#             break
#         elif i !=completion[0]:
#             answer=i
#             break
#         else:
#             completion.pop(0)
#     return answer